### PR TITLE
ci: fix googleapis change propagation job

### PR DESCRIPTION
**Pull request by suztomo**
Fixed pull request creation logic to use yohsi-code-bot.

### CODE REVIEW

File: templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml [hunk 1]
```diff
@@ -9,7 +10,42 @@ jobs:
     - uses: actions/checkout@v3
       with:
         repository: 'googleapis/google-cloud-java'
-    - run: |
-        .kokoro/sync_with_googleapis.sh monorepo_script_output
+    - name: Setup workspace
+      run: |
+        mkdir -p "${WORKSPACE}"
+        git config user.email "yoshi-code-bot[bot]@users.noreply.github.com"
```
Comment (meltsufin) [inline templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:16]: Why are you calling it yoshi-code-bot?

File: templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml [hunk 2]
```diff
@@ -9,7 +10,42 @@ jobs:
...
+      working-directory: /tmp/googleapis-sync/google-cloud-java
+    - name: Create pull request
+      run: |
+        gh pr create --title='chore: googleapis change propagation via hermetic build' \
+            --body='googleapis change propagation via hermetic build' \
+            --base="${BASE_BRANCH}" \
+            --head=monorepo_googleapis_change
+      working-directory: /tmp/googleapis-sync/google-cloud-java
+      env:
+        GITHUB_TOKEN: ${{ secrets.YOSHI_CODE_BOT_TOKEN }}
```
Comment (meltsufin) [inline templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:49]: Can you use built-in `GITHUB_TOKEN` instead?

File: templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml (hunk 1 above)
Comment (suztomo) [reply templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:16]: It's for consistency with existing job that creates pull requests https://github.com/googleapis/google-cloud-java/blob/main/.github/workflows/readme.yaml#L34

**Comment by suztomo** [comment]
@meltsufin PTAL

File: templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml (hunk 1 above)
Comment (meltsufin) [reply templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:16]: Yoshi is a defunct term. We probably need a new convention.

File: templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml (hunk 2 above)
Comment (meltsufin) [reply templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:49]: Do we need to create forks?

Comment (suztomo) [reply templates/.github/rename-to-workflows/googleapis_hermetic_sync.yaml:49]: No, we do not need to create forks as of now. Inability of forking is just an example limitation I found using the default token (https://github.com/googleapis/google-cloud-java/pull/7931#issuecomment-1163771450).

**Comment by suztomo** [comment]
The default token fails with CLA check. Other than that, it works.

Screen Shot 2022-06-27 at 5 01 53 PM

**Comment by suztomo** [comment]
@meltsufin Now it uses the default token. PTAL.

**Review by meltsufin — APPROVED** [review]

**Comment by meltsufin** [comment]
For the CLA, can we just put in an `@google.com` email for the commit author?

**Comment by suztomo** [comment]
The pull request author needs to be signed with CLA. Automating pull request creation requires an automation account.

**Pull request merged**